Fix several issues with Quick Unlock (#9697)
keepassxreboot/keepassxc: docs/topics/DatabaseOperations.adoc

=== Quick Unlock
On Windows and macOS, subject to hardware availability, your credentials can be securely stored to enable subsequent unlocking of your database through biometric authentication. This is enabled by default on Windows using _Windows Hello_ and on macOS using _Touch ID or Apple Watch_ services. You can disable this feature in the Application Settings under the Security section.

> NOTE: On Windows, you will be prompted to authenticate to Windows Hello after unlocking your database with full credentials. This is required to access the hardware certificate store that secures Quick Unlock. If you cancel this prompt, your database will continue to unlock, but Quick Unlock will not be enabled.

Reviewer: I think this would be clearer:
```suggestion
NOTE: On Windows, you will be prompted to authenticate to Windows Hello after unlocking your database with full credentials. This is required to store your credentials in a secure hardware store. If you cancel this prompt, your database will continue to unlock, but Quick Unlock will not be enabled.
```
Primarily, "access" can be misinterpreted as a retrieval. Though I admit either version is a bit too technical. Maybe just "... to configure QuickUnlock. ..."

Author: However, it is completely false. We don't store credentials in a hardware store. The certificate used to derive key material is stored in the hardware store, you need to authenticate to use/access that certificate.

Reviewer: You store _something_ in there, don't you? (Encrypted credentials? Some identifier? I don't remember the details, but that's less important). Otherwise you wouldn't need that prompt.

Anyway, maybe just "This is required to configure QuickUnlock."

Author: No, nothing is stored by us in the TPM. How all this works is a certificate is generated by Windows from the TPM root certificate. I'm fairly certain this certificates private key is stored in the TPM, but it could also just be encrypted by the TPM root cert and really stored in windows software credential store. Anyway, you need to authenticate to use the TPM (ie make the kernel calls) to access this cert to sign the challenge and generate the key material that is used to encrypt/decrypt your credentials.

Reviewer: And then those encrypted credentials still need to be stored somewhere? So that they would be available to unlock the database later.

Anyway, those technical details are besides the point. I still think that "access the hardware certificate store that secures Quick Unlock" is too technical.

Author: Yah the encrypted credentials are stored in keepassxc memory. Check out my update, that should be clear

Reviewer: Looks good.
